chain/beacon/store.go

```diff
@@ -79,7 +80,7 @@ func (a *schemeStore) Put(ctx context.Context, b *chain.Beacon) error {
 	// If the scheme is unchained, previous signature is set to nil. In that case,
 	// relationship between signature in the previous beacon and previous signature
 	// on the actual beacon is not necessary. Otherwise, it will be checked.
-	if a.sch.DecouplePrevSig {
+	if !a.sch.IsPreviousSigSignificant {
```

Comment:
Agreeing with other comments from Patrick: having a `scheme.Verify(beacon)` and `beacon.isNext(previous,current)` would remove such logic.

Reply:
I'll explore that in a next commit hopefully this week